src/bin/path_filestat.rs

```diff
@@ -129,6 +129,30 @@ unsafe fn test_path_filestat(dir_fd: wasi_unstable::Fd) {
     assert_eq!(stat.st_mtim, new_mtim, "mtim should not change");
     assert_eq!(stat.st_atim, old_atim, "atim should not change");
 
+    let new_atim = old_atim - 100;
+    assert_eq!(
+        wasi_path_filestat_set_times(
+            dir_fd,
+            0,
+            filename,
+            new_atim,
+            new_atim,
+            wasi_unstable::FILESTAT_SET_ATIM | wasi_unstable::FILESTAT_SET_ATIM_NOW,
+        ),
+        Err(wasi_unstable::EINVAL),
+        "ATIM & ATIM_NOW can't both be set"
+    );
+
+    // check if the times were untouched
+    let status = wasi_path_filestat_get(dir_fd, 0, filename, filename.len(), &mut stat);
+    assert_eq!(
+        status,
+        wasi_unstable::raw::__WASI_ESUCCESS,
```

Comment:
Or `wasi_unstable::ESUCCESS`?

Reply:
Alas, in `wasi_unstable` there is no `ESUCCESS` as such. Instead, if a call returns a `Result` a unit tuple corresponds to `ESUCCESS`. See [here](https://github.com/CraneStation/rust-wasi/blob/master/src/wasi_unstable/mod.rs#L261).

Reply:
Fair enough. we could consider exposing ESUCCESS in wasi-unstable if we find another use than tests

Reply:
Hmm, should we though? I think it's enough to have it inside a submodule as is now: `wasi_unstable::raw::__WASI_ESUCCESS`, don't you think? On the other hand, I would also like it if we could refer to it simply as `wasi_unstable::ESUCCESS`. All in all, I'm OK either way :-) @sunfishcode thoughts?

Reply:
The errno values in `wasi_unstable` are of type `NonZeroU16`, and `ESUCCCESS` is 0 ;-). We could have an `ESUCCESS` in `wasi_unstable`, but it'd be confusing because it'd be a different type than the other `E*` values.

The underlying problem is that we want to test that out parameters in WASI functions are actually stored to. In the `wasi` crate's Rust-idiomatic API, it translates out parameters to additional return values within a `Result`, which is almost always what actual application or library code would want, but it doesn't let us pass in a reference and then test that the reference was stored to. So for functions with out parameters.

How about this: We could make `wasi_path_filestat_get` return a `Result<(), wasi_unstable::Error>` instead of an `wasi_unstable::Errno`. The regular return value would still be returned through an out parameter, but we'd still get a `Result` so we don't have to deal with `ESUCCESS` explicitly. Thoughts?